apps/examples/csifw_test/csifw_test.c

```diff
@@ -0,0 +1,214 @@
+ ****************************************************************************/
+
+#include "csifw_test.h"
+#include "csifw/csifw_api.h"
+
+#include <stdio.h>
+
+/****************************************************************************
+ * csifw_test
+ ****************************************************************************/
+
+#include <fcntl.h>
+static int gFd; /* File descriptor*/
+
+#define PRNT_LN1 printf("______________________________________________________________________________________________________\n")
+#define PRNT_LN2 printf("______________________________________________________________________________________________________\n\n")
+#define PRNT_LN3 printf("\n\n______________________________________________________________________________________________________\n")
+
+
+static unsigned int gCounter;
+
+static void print_buf(const unsigned char* buf, int len)
+{
+	printf("\n RAW DATA %d\n\n", len);
+	unsigned long long *buff_tmp = (u64 *)buf;
+	int buff_len = (len/8) + 1;
+	for (int i = 0; i < buff_len; i++) 
+		printf("[%02d]0x%016llx\n", i, buff_tmp[i]);
+}
+
+static void demo_upd_raw_data_listener(CSIFW_RES res, int csi_buff_len, unsigned char *csi_raw_buff, void* ptr)
+{
+	if(res == CSIFW_OK) {
+		gCounter++;
+		printf("\n%d. Raw Data received in APP [%d]\n", gCounter, csi_buff_len);
+		printf("\n[APP]LOGGING DATA\n");
+		print_buf(csi_raw_buff, csi_buff_len);
+		return;
+	}
+	if(res == CSIFW_ERROR_WIFI_DIS_CONNECTED) {
+		printf("<URGENT CALLBACK N/W CHANGE>[APP] wifi disconnected\n");
+		printf("<URGENT CALLBACK N/W CHANGE>[APP] so service will PAUSE\n");
+		return;
+	}
+	if (res == CSIFW_ERROR) {
+		printf("[APP] Some error\n");
+		return;
+	}
+
+}
+
+#define LOG_PARSED_DATA 1
+static void demo_upd_parsed_data_listener(CSIFW_RES res, int csi_data_len, float *csi_parsed_buff, void* ptr)
+{
+	// print parsed data buffer
+	printf("\n%d. Parsed Data received in APP [%d]\n", gCounter, csi_data_len);
+	
+	#if LOG_PARSED_DATA
+	printf("[APP] Displaying parsed data, csi_data_len: %d\n\n", csi_data_len);
+	for(int i = 0; i < csi_data_len; i += 2) {
+		printf("sub_carrier:%d[%f,%f], ", i/2, csi_parsed_buff[i], csi_parsed_buff[i+1]);
+		}
+		printf("\n");
+	#endif
+	printf("\n\n");
+}
+
+int csifw_test_main(void)
+{
+	printf("\n[APP]: CSI Manager DEFAULT FLOW \n");
+	PRNT_LN1;
+	PRNT_LN2;
+
+	PRNT_LN3;
+	printf("[APP]: csi_manager_init\n");
+	csi_manager_init();
+	PRNT_LN3;
+
+	printf("[APP]: csi_service_init\n");
+	PRNT_LN2;
+	CSIFW_RES res = csi_service_init(demo_upd_raw_data_listener, demo_upd_parsed_data_listener, 40000, NULL); 
+	if (res != CSIFW_OK) {
+		PRNT_LN2;
+		if (res == CSIFW_ERROR_WIFI_NOT_CONNECTED) {
+			printf("\n[APP]: CSIFW-> {WIFI NOT CONNECTED}\n");
+		} else {
+			printf("\n[APP]: CSI Manager INIT FAIL\n");
+			return 0;
+		}
+	}
+	PRNT_LN2;
+	printf("\n[APP]: CSI Manager INIT SUCCESS\n");
+	printf("[APP]: SLEEP--> 5 seconds\n");
+	PRNT_LN2;
+	sleep(5);
+
+	PRNT_LN3;
+	csi_action_param_t config = {0};
```

Comment:
I think most of app developer can't understand what this config means exactly.
so please refer to below.
```
some_csi_init_in_framework()
{
	/* Initialize Base value Here */
	g_config.group_num = CSI_GROUP_NUM_1;
	g_config.mode = CSI_MODE_NORMAL;
	g_config.accuracy = CSI_ACCU_1BYTE;
	g_config.ch_opt = CSI_CH_NON_LEGACY;
	g_config.trig_period = 200;
	g_config.data_rate = CSI_DATA_RATE_HIGH...; //0x80 is impossible to understand what it means.
}

csi_set_group_num(int num)
{
	if (state_is_wrong) {
		return CSIFW_ERROR_STATE;
	}
	g_config.group_num = num;
}
```
A better way is that structure in driver need to be avoided, so it should work without csi_service_set_config based on basic setting value.

Reply:
Actually for the application to get data, they must configure what specific kind of csi data is required.

with `some_csi_init_in_framework()` do you wish to provide option to use default config for an app-developer who does not have info about config? And then provide apis (like `csi_set_group_num()` )for every config component, so that they can only change the config they are known too?